oxide/lib_test.go

```diff
@@ -144,3 +155,58 @@ func Test_buildRequest(t *testing.T) {
 		})
 	}
 }
+
+func Test_Client(t *testing.T) {
+	tt := map[string]struct {
+		config    *Config
+		env       map[string]string
+		wantError bool
+	}{
+		"valid client from config": {
+			config: &Config{
+				Address: "http://localhost",
+				Token:   "foo",
+			},
+		},
+		"valid client from env": {
+			env: map[string]string{
+				"OXIDE_HOST":  "http://localhost",
+				"OXIDE_TOKEN": "foo",
+			},
+		},
+		"missing address": {
+			config: &Config{
+				Token: "foo",
+			},
+			wantError: true,
+		},
+		"missing token": {
+			config: &Config{
+				Address: "http://localhost",
+			},
+			wantError: true,
+		},
+	}
+
+	for testName, testCase := range tt {
+		t.Run(testName, func(t *testing.T) {
+			for key, val := range testCase.env {
+				os.Setenv(key, val)
+			}
+
+			t.Cleanup(func() {
+				for key := range testCase.env {
+					os.Unsetenv(key)
+				}
+			})
+
+			_, err := NewClient(testCase.config)
+			if testCase.wantError {
+				assert.Error(t, err, "")
+			} else {
+				assert.NoError(t, err, "")
+			}
```

Comment:
Let's use `assert.Equal` to verify that the contents of the errors and the Client are what we expect them to be.
Something like:
```suggestion
			got, err := NewClient(testCase.config)
			assert.Equal(t, tt.want, got)
			assert.Equal(t, tt.wantErr, err)
```

Reply:
I ended up adding `expectedClient` and `expectedError` fields to the struct so I could asset on the returned `Client` and `error`. Let me know what you think there.

Reply:
Looks good thanks!